rules/prefer-promise-static-methods.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Rule: prefer-promise-static-methods
+ * Prefer `Promise.resolve(foo)` to `new Promise((resolve) => resolve(foo))`.
+ */
+
+'use strict'
+
+const { getSourceCode } = require('./lib/eslint-compat')
+const getDocsUrl = require('./lib/get-docs-url')
+const {
+  isPromiseConstructorWithInlineExecutor,
+} = require('./lib/is-promise-constructor')
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Prefer `Promise.resolve(foo)` to `new Promise((resolve) => resolve(foo))`.',
+      url: getDocsUrl('prefer-promise-static-methods'),
+    },
+    messages: {
+      replaceWithStaticMethod:
+        'Prefer `Promise.{{ method }}()` to `new Promise()`. The static method is faster, more readable and less verbose.',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+  create(context) {
+    const sourceCode = getSourceCode(context)
+    /**
+     * Report an error if the given node is a call to `resolve` or `reject`.
+     * @param {import('estree').SimpleCallExpression} callNode
+     * @param {import('estree').NewExpression} constructorNode
+     * @param {Array<string | undefined>} parameterNames
+     */
+    function reportIfIsPromiseCall(callNode, constructorNode, parameterNames) {
+      if (
+        callNode.callee.type === 'Identifier' &&
+        callNode.arguments.length <= 1
```

Comment:
What about `new Promise(resolve => resolve(itCanThrowAnError()))` (the case of [`Promise.try`](https://github.com/tc39/proposal-promise-try))?

Reply:
Hmm I'm not sure what the best approach there would be. (Also I didn't realize `new Promise` is same tick, I must've misread the spec.) Perhaps just show a suggestion instead of an autofix?

Reply:
I think that cases that could throw an error should be ignored (however, it's too many possible cases). Also, can be added an option to check such cases.